File: src/so/aporia/utils/user/command/impl/HelpCommand.java
package so.aporia.utils.user.command.impl;

import so.aporia.utils.user.command.Command;
import so.aporia.utils.user.command.CommandManager;

public class HelpCommand implements Command {

    @Override
    public String name() { return "help"; }

    @Override
    public String description() { return "Lists all available commands"; }

    @Override
    public void execute(String[] args) {
        CommandManager.chat("§6--- Aporia Commands ---");
        CommandManager.INSTANCE.getAll().stream()
            .sorted((a, b) -> a.name().compareTo(b.name()))
            .forEach(cmd -> CommandManager.chat("§e" + CommandManager.PREFIX + cmd.name() + " §7- " + cmd.description()));
    }
}

File: src/so/aporia/utils/user/command/impl/HelpCommand.kt
package so.aporia.utils.user.command.impl

import com.chaos.annotation.Obfuscate
import so.aporia.utils.user.command.Command
import so.aporia.utils.user.command.CommandManager

@Obfuscate
class HelpCommand : Command {
    override fun name() = "help"
    override fun description() = "Lists all available commands"
    override fun execute(args: Array<String>) {
        CommandManager.chat("§6--- Aporia Commands ---")
        CommandManager.getAll().sortedBy { it.name() }.forEach { cmd ->
            CommandManager.chat("§e${CommandManager.PREFIX}${cmd.name()} §7- ${cmd.description()}")
        }
    }
}
